app/src/main/java/org/jellyfin/androidtv/auth/repository/ServerRepository.kt

```diff
@@ -66,15 +68,17 @@ class ServerRepositoryImpl(
 	}
 
 	override suspend fun loadDiscoveryServers() {
-		val servers = mutableListOf<Server>()
-
-		jellyfin.discovery
-			.discoverLocalServers()
-			.map(ServerDiscoveryInfo::toServer)
-			.collect { server ->
-				servers += server
-				_discoveredServers.emit(servers.toList())
-			}
+		withContext(Dispatchers.IO) {
```

Comment:
Use `viewModelScope.launch(Dispatchers.IO)` in the StartupViewModel instead.

Reply:
um i have no clue how to get `startupViewModel` from the ServerRepository scope?

Reply:
No I mean to replace line 73 in the viewmodel (`viewModelScope.launch { serverRepository.loadDiscoveryServers() }`) with `viewModelScope.launch(Dispatchers.IO) { serverRepository.loadDiscoveryServers() }` so you don't need the `withContext(){}` here.

Reply:
ah lol, yeah makes more sense.